### PR TITLE
Add possibility to apply a filter to LogCaptor's logs

### PR DESCRIPTION
**Pull request by sleepo581**
Hi! I have found your library very useful. 
I'm working on a project, where logs have business meaning (some of them are even persisted in database, some of them act like alerts for monitoring team and etc ...).
We created our own appenders  and applied some custom filters to filter out business logs from regular application logs.

I think we could use your library for testing purposes, however it would be nice, if the library would support logs filtering.

For ex. we want to capture all the logs related to user creation and don't gather unwanted logs. We would do something like this (assuming we don't know exact class for this log event):
```java
public class UserCreationFilter extends Filter<ILoggingEvent> {

    @Override
    public FilterReply decide(ILoggingEvent event) {
        if (event.getMarker() == USER_ACTIONS && isNotEmpty(event.getArgumentArray())) {
            return event.getArgumentArray()[0] == CREATED ? FilterReply.ACCEPT : FilterReply.DENY;
        }
        
        return FilterReply.DENY;
    }
}
```
```java
LogCaptor captor = LogCaptor.forName("org.organization");
captor.setFilter(new UserCreationFilter()); 
```

Without filter it will capture the whole application logs, but with the filter it will reduce the amount of captured logs.

### CODE REVIEW

File: src/main/java/nl/altindag/log/LogCaptor.java [hunk 1]
```diff
@@ -140,6 +141,11 @@ public static LogCaptor forName(String name) {
                 .collect(toList());
     }
 
+    public void addFilter(Filter<ILoggingEvent> filter) {
```
Comment (Hakky54) [inline src/main/java/nl/altindag/log/LogCaptor.java:144]: The addFilter method is between `getLogEvents` and `toLogEvent`. getLogEvents method uses toLogEvent method and therefore I prefer to keep them together. Would it possible for you to move the addFilter method below toLogEvent?

**Comment by Hakky54** [comment]
Hi @sleepo581 

Thank you for taking your time and effort to create this cool PR! I like the idea of adding this kind of extension to the library to give the end-user more flexibility for their use case such as filtering. I have dropped a minor feedback, could you have a look at it?

File: src/main/java/nl/altindag/log/LogCaptor.java (hunk 1 above)
Comment (sleepo581) [reply src/main/java/nl/altindag/log/LogCaptor.java]: 👍 Fixed

Comment (Hakky54) [reply src/main/java/nl/altindag/log/LogCaptor.java]: Thank you very much for quickly adjusting it

**Comment by sleepo581** [comment]
> Hi @sleepo581
> 
> Thank you for taking your time and effort to create this cool PR! I like the idea of adding this kind of extension to the library to give the end-user more flexibility for their use case such as filtering. I have dropped a minor feedback, could you have a look at it?

I've forced pushed a new version of branch with minor fixed. Thanks for review!

**Comment by Hakky54** [comment]
Looks good, thank you again for adding this feature, I merged it and I will also release a new version today

**Pull request merged**